src/core/lib/transport/transport.h

```diff
@@ -228,6 +229,61 @@ struct CallArgs {
 using NextPromiseFactory =
     std::function<ArenaPromise<ServerMetadataHandle>(CallArgs)>;
 
+// TODO(ctiller): eventually drop this when we don't need to reference into
+// legacy promise calls anymore
+class CallSpineInterface {
```

Comment:
Please add a comment explaining the purpose of this class and its relationship to `CallInitiator`, `CallHandler`, top-level calls, client_channel calls, and server calls.  Also its importance for cancellation.

Reply:
done